lib/firebase_functions_interop.dart

```diff
@@ -378,3 +381,51 @@ class FirestoreEvent extends Event<DeltaDocumentSnapshot> {
           timestamp: timestamp,
         );
 }
+
+class PubsubFunctions {
+  const PubsubFunctions._();
+
+  TopicBuilder topic(String path) => new TopicBuilder._(_js.pubsub.topic(path));
+}
+
+class TopicBuilder {
+  @protected
+  final js.TopicBuilder nativeInstance;
+
+  TopicBuilder._(this.nativeInstance);
+
+  /// Event handler that fires every time an event is public in Pubsub.
+  js.CloudFunction onPublish(FutureOr<void> handler(Message message)) {
+    dynamic wrapper(js.Message jsMessage) => _handleEvent(jsMessage, handler);
+    return nativeInstance.onPublish(allowInterop(wrapper));
+  }
+
+  dynamic _handleEvent(
+      js.Message jsMessage, FutureOr<void> handler(Message message)) {
+    final Message message = new Message(jsMessage);
+    var result = handler(message);
+    if (result is Future) {
+      return futureToPromise(result);
+    }
+    return null;
+  }
+}
+
+class Message {
+  Message(js.Message this.nativeInstance);
+
+  @protected
+  final js.Message nativeInstance;
+
+  /// User-defined attributes published with the message, if any.
+  Map<String, String> get attributes => nativeInstance.attributes;
```

Comment:
This probably need to convert JS object into a Dart Map with dartify:

```dart
Map<String, dynamic> get attributes => dartify(nativeInstance.attributes);
```

Also see my other comment about this field in the interop class.